ORK1Kit/ORK1Kit/ActiveTasks/CEV1RangeOfMotionStepViewController.swift

```diff
@@ -0,0 +1,144 @@
+            rawAngle = (pitchAngle > 0) ? pitchAngle : 180 + (180 + pitchAngle)
+        case .roll:
+            let rollAngle = Measurement(value: atan2(attitude.rotationMatrix.m31, attitude.rotationMatrix.m11), unit: UnitAngle.radians).converted(to: .degrees).value
+            rawAngle = (rollAngle > 0) ? rollAngle : 180 + (180 + rollAngle)
+        case .yaw:
+            let yawAngle = Measurement(value: atan2(attitude.rotationMatrix.m21, attitude.rotationMatrix.m11), unit: UnitAngle.radians).converted(to: .degrees).value
+            rawAngle = (yawAngle > 0) ? yawAngle : 180 + (180 + yawAngle)
+        }
+        
+        if let lastRawAngle {
+            let correctedLastRawAngle = lastRawAngle + Double(clockwiseCrossesZero) * 360
+            if crossesAxis(rawAngle: rawAngle, lastRawAngle: lastRawAngle) {
+                if debugMode {
+                    print("Axis Crossed - cumulativeAngle: \(cumulativeAngle); firstRawAngle: \(firstRawAngle)")
+                }
+                switch directionOfRotation(rawAngle: rawAngle, lastRawAngle: lastRawAngle) {
+                case .clockwise:
+                    clockwiseCrossesZero += 1
+                case .counterClockwise:
+                    clockwiseCrossesZero -= 1
+                }
+            }
+            cumulativeAngle += (rawAngle + Double(clockwiseCrossesZero) * 360 - correctedLastRawAngle)
+        }
+        
+        guard motionUpdates > 1 else {
+            return nil
+        }
+        if motionUpdates == 2 {
+            firstRawAngle = rawAngle
+            if debugMode {
+                print("firstRawAngle: \(firstRawAngle)")
+            }
+        }
+        lastRawAngle = rawAngle
+            
+        switch step.measuredDirectionOfRotation {
+        case .clockwise:
+            return cumulativeAngle + firstRawAngle - step.offsetFromReferenceAxis.converted(to: .degrees).value
+        case .counterClockwise:
+            return -(cumulativeAngle + firstRawAngle - step.offsetFromReferenceAxis.converted(to: .degrees).value)
+        }
+        
+    }
+    
+    // assumes maximum rotation rate < 10 degrees per sample, at 100 Hz = 1000 degrees / sec, or 2.8 rotations per second ~ 167 rpm, probably not physically possibly by a human doing normal rotation of a joint
+    func directionOfRotation(rawAngle: Double, lastRawAngle: Double) -> CEV1DirectionOfRotation {
+        if crossesAxis(rawAngle: rawAngle, lastRawAngle: lastRawAngle) {
+            return (rawAngle < lastRawAngle) ? .clockwise : .counterClockwise
+        } else {
+            return (rawAngle > lastRawAngle) ? .clockwise : .counterClockwise
+        }
+    }
+    
+    // assumes maximum rotation rate < 10 degrees per sample, at 100 Hz = 1000 degrees / sec, or 2.8 rotations per second ~ 167 rpm, probably not physically possibly by a human doing normal rotation of a joint
```

Comment:
So the reason this number is not the smallest possible has to do with observances when crossing the reference axis. This can be demonstrated by running the **CEVRKUITestVehicle [serial]** schema and running it on a device. Set the `debugMode` variable in the `CEV[1]RangeOfMotionStepViewController` to `true` so it will spit out state to the console in Xcode. Select the `RangeOfMotionStep` from the list. In this test, it would expected that the phone begins -90 degrees from the reference axis of "flat on a table". Now if you run this test and move the phone thru the 0-pitch reference axis, you will see a change in the number of axis crossings reported. Where things get weird is if you cross that line very quickly and changing direction. So think of using your phone like a fan and rapidly pitching it up and down across the axis. If we're tracking it correctly, since we're never doing a full rotation, the number should fluctuate between -1 and 0, or 1 and 0 depending on measured direction of rotation. In this use case, I note many more erroneous measurements when I use 5 vs 10. Now whether or not this will make a difference in real world use cases? 🤷‍♂️. Perhaps we set a default value but allow one to be passed as a parameter on the step so the researcher can play with it?

Reply:
I think I understand this better now, thanks! Sounds like perhaps we could leave this as-is, and wait to parameterize or adjust in the future if we have a more specific scenario to address?